local_planner/src/nodes/star_planner.cpp

```diff
@@ -109,6 +110,27 @@ float StarPlanner::treeHeuristicFunction(int node_number) const {
 
 void StarPlanner::buildLookAheadTree() {
   std::clock_t start_time = std::clock();
+
+  int origin = 0, old_origin = 0;
```

Comment:
Hello

Having all of the variables declared before they are used doesn't actually improve performance, especially for things like `int`, `float` and other types that don't use heap allocation. Modern compilers will correctly allocate the space for the variable outside of the loop if it will cause a speedup, or not if it will slow down the code.

Especially with C++, it is important to keep the code more readable, and let the compiler do the optimization work.

Could you then put the variable declarations back to where they are used? The only places I see it making sense to declare at the front is something like `std::vector` or `Eigen::Matrix`, if it will be resized multiple times during the lifetime of the variable.

Reply:
Done